### ISSUE TITLE
💅 `useExhaustiveDependencies` doesn't catch missing function dependencies even when they are written as regular functions

### ISSUE THREAD

**Issue by misicnenad** [biomejs/biome#2362]
### Environment information

```bash
CLI:
  Version:                      1.5.3
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           linux

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           "v19.6.0"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "yarn/4.1.1"

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    false
  VCS disabled:                 true

Workspace:
  Open Documents:               0
```

### Rule name

useExhaustiveDependencies

### Playground link

https://biomejs.dev/playground/?lintRules=all&code=aQBtAHAAbwByAHQAIAB7ACAAdQBzAGUARQBmAGYAZQBjAHQALAAgAHUAcwBlAFMAdABhAHQAZQAgAH0AIABmAHIAbwBtACAAIgByAGUAYQBjAHQAIgA7AAoACgBlAHgAcABvAHIAdAAgAGYAdQBuAGMAdABpAG8AbgAgAEYAYQB1AGwAdAB5AEwAaQBuAHQAZQByACgAKQAgAHsACgAgACAAYwBvAG4AcwB0ACAAWwB2AGEAbAB1AGUALAAgAHMAZQB0AFYAYQBsAHUAZQBdACAAPQAgAHUAcwBlAFMAdABhAHQAZQAoACIAIgApAAoAIAAgAAoAIAAgAGYAdQBuAGMAdABpAG8AbgAgAGYAdQBuAGMAKAApACAAewAKACAAIAAgACAAcwBlAHQAVgBhAGwAdQBlACgAIgBzAG8AbQBlACIAKQAKACAAIAB9AAoAIAAgAAoAIAAgAHUAcwBlAEUAZgBmAGUAYwB0ACgAKAApACAAPQA%2BACAAewAKACAAIAAgACAAZgB1AG4AYwAoACkAIAAgAAoAIAAgAH0ALAAgAFsAXQApAAoAIAAgAAoAIAAgAHIAZQB0AHUAcgBuACAAPAA%2BAHsAdgBhAGwAdQBlAH0APAAvAD4ACgB9AAoACgBlAHgAcABvAHIAdAAgAGYAdQBuAGMAdABpAG8AbgAgAFcAbwByAGsAaQBuAGcATABpAG4AdABlAHIAKAApACAAewAKACAAIABjAG8AbgBzAHQAIABbAHYAYQBsAHUAZQAsACAAcwBlAHQAVgBhAGwAdQBlAF0AIAA9ACAAdQBzAGUAUwB0AGEAdABlACgAIgAiACkACgAgACAACgAgACAAYwBvAG4AcwB0ACAAZgB1AG4AYwAgAD0AIAAoACkAIAA9AD4AIAB7AAoAIAAgACAAIABzAGUAdABWAGEAbAB1AGUAKAAiAHMAbwBtAGUAIgApAAoAIAAgAH0ACgAgACAACgAgACAAdQBzAGUARQBmAGYAZQBjAHQAKAAoACkAIAA9AD4AIAB7AAoAIAAgACAAIABmAHUAbgBjACgAKQAgACAACgAgACAAfQAsACAAWwBdACkACgAgACAACgAgACAAcgBlAHQAdQByAG4AIAA8AD4AewB2AGEAbAB1AGUAfQA8AC8APgAKAH0A

### Expected result

It should catch missing function dependencies both when the function is written as a regular function and when it's written as an arrow function.

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct
